docs/Frontends.md

```diff
@@ -0,0 +1,84 @@
+
+Unfortunately this creates a bit of a barrier for organizations evaluating such components. Please don't hesitate to join the OpenTripPlanner Gitter chat and ask if you need any information on this subject. Several organizations that manage multiple large OTP deployments are active there and will be happy to engage with you.
+
+While you are welcome to stand up an instance of one of these UIs for testing or evaluation, and there has been some effort to make this more straightforward especially on the Arcadis IBI projects, it is still strongly recommended to collaborate with a specialist. The process of tailoring a deployment to a specific region and deployment environment is estimated to take at least several full days even for an experienced professional, and while it may start off looking deceptively simple, can rapidly grow into a seemingly never-ending project for those who are not aware of the pitfalls. The goal here is not by any means to discourage you from using or learning about any of these projects, but to set expectations that the process will be much smoother if you work with someone who's done it before.
+
+If you do decide to take on this task, be aware that both of these systems rely on multiple additional backend services other than the OTP Java backend. To get either one working, you will need to configure a source of map tiles and a geocoder that translates geographic names to latitude and longitude coordinates, among other components.
+
+### Evaluating OTP-React-Redux
+
+At the time of this writing (early 2024) it is possible to try out the OTP-React-Redux production frontend maintained by Arcadis IBI with only a few adjustments to the example configuration file, so long as you are working on a **non-production test deployment**. To do so, you can use the map tile sets pre-configured in the map.baseLayers section of its `example-config.yml`, along with a test account from the maintainers of the Pelias geocoder at https://geocode.earth. They provide a free temporary geocoding account lasting about two weeks, which should be sufficient for testing out the system. You'll need to substitute their Pelias server URL and your temporary API key into the OTP-React-Redux `example-config.yml` in the geocoder.apiKey and geocoder.baseUrl sections. You'll also want to update the geocoder boundary rectangle and focusPoint.
```

Comment:
I must admit I'm quite reluctant to include quite specific installation instructions for another piece of Software in the OTP docs as they are very likely to be out of date soon.

Reply:
I understand, I was of course thinking the same thing. Maybe I should instead coordinate to get this information into the OTP-RR repo. However, we should establish whether this OpenTripPlanner documentation is strictly about the backend, or about the whole OpenTripPlanner project. OTP-RR might have some kind of special status among frontends. It is unique in being under the OpenTripPlanner organization (I believe this was a decision of the OTP project steering committee). My sense is that at least back when David and Landon at Conveyal were working on this, it was just seen as a new default UI for OTP, and it may still effectively serve that purpose given its current ease of configuration.

Reply:
I think it's good to have some general information but I fear that reference to specific config values and step-by-step instructions will be so quickly out of date that they are more trouble than they are worth. This is particularly problematic because once we release these instructions will be on the website forever.

Reply:
I would be less concerned if this wasn't deployed to the docs website and we could find a way to somehow maintain it elsewhere, perhaps even in the OTP repo itself (but not the website).

Reply:
Point taken, this is really just a case of the README/docs/config defaults of another repo needing a small refresh. I'll pursue that path and see if I can reduce this section down to a single cross-project reference.

Reply:
> I would be less concerned if this wasn't deployed to the docs website and we could find a way to somehow maintain it elsewhere, perhaps even in the OTP repo itself (but not the website).

There are conflicting goals here though. For many people just approaching or testing out OTP, it is not at all obvious how to get a "real" (non-debug) frontend working. Even the last time you and I were talking about it, we were unsure if it would take days/weeks for someone experienced. In fact, a minimal working instance can be had in a couple of minutes if you know which 3 lines to change in the config. This information needs to be very publicly visible (and discoverable by search) if the intent is to draw people to the project and make it approachable.

Reply:
PR at opentripplanner/otp-react-redux#1138

Reply:
I find this a much better solution. Thanks.

Reply:
> However, we should establish whether this OpenTripPlanner documentation is strictly about the backend...

My impression is that the [www.opentripplanner.org](https://www.opentripplanner.org/) is all about the OTP server. But, this should be at leased dicussed by the PLC, and if so there will be a note of it in the minutes - witch is enough for me.

Reply:
I added this to the agenda for the next PLC meeting https://docs.google.com/document/d/1uuNGUNBaA3MTY-UVEb5Tt1lpqlerJVcsjmchEqV7UF8/edit?usp=sharing

Reply:
Thanks. Even if these published OTP docs don't cover details of frontend configuration / installation, do you consider something like the rest of this changeset to be appropriate? If we don't have a high-level page like this, how would readers become aware of the main frontend options and how they relate to this backend?

Reply:
I am in favour of keeping the official doc focused on OTP, but having docs like this one explaining the integration points is good. In the future I would like to see a more complete recommendation for the whole stack of components in a travel planner ecosystem - there are some progress made on this the last year, but we have a long way to go. Where this should be is another question.